opba-protocols/sandboxes/hbci-sandbox/src/main/java/de/adorsys/opba/protocol/sandbox/hbci/protocol/authenticated/nonauthorized/AuthenticatedCustomMsg.java

```diff
@@ -34,26 +35,32 @@ protected String getTemplatePathAndUpdateCtxIfNeeded(HbciSandboxContext context)
             if (context.getBank().getSecurity().getTransactions() == SensitiveAuthLevel.AUTHENTICATED) {
                 return "response-templates/authenticated/custom-message-konto-mt940.json";
             }
-
             if (RequestStatusUtil.isForTransactionListing(context.getRequestData())) {
                 context.setAccountNumberRequestedBeforeSca(MapRegexUtil.getDataRegex(context.getRequestData(), "TAN2Step6\\.OrderAccount\\.number"));
             }
-
             return getAuthorizationRequiredTemplateOrWrongTanMethod();
         }
 
         if (context.getRequestData().keySet().stream().anyMatch(it -> it.startsWith(PAYMENT))) {
             if (context.getBank().getSecurity().getPayment() == SensitiveAuthLevel.AUTHENTICATED) {
                 return "response-templates/authenticated/custom-message-konto-mt940.json";
             }
-
             if (RequestStatusUtil.isForPayment(context.getRequestData())) {
                 context.setAccountNumberRequestedBeforeSca(MapRegexUtil.getDataRegex(context.getRequestData(), "TAN2Step6\\.OrderAccount\\.number"));
             }
-
             return "response-templates/authenticated/custom-message-authorization-required-payment.json";
         }
 
+        if (context.getRequestData().keySet().stream().anyMatch(it -> it.startsWith(PAYMENT_STATUS))) {
+            if (context.getBank().getSecurity().getPaymentStatus() == SensitiveAuthLevel.AUTHENTICATED) {
+                return "response-templates/authenticated/custom-message-payment-status.json";
+            }
+            if (RequestStatusUtil.isForPaymentStatus(context.getRequestData())) {
+                context.setAccountNumberRequestedBeforeSca(MapRegexUtil.getDataRegex(context.getRequestData(), "TAN2Step6\\.OrderAccount\\.number"));
```

Comment:
Probably, we don't want to set account number here (is it even available?), but rather some payment identifier

Reply:
now it never comes to this line of code